modules/javafx.graphics/src/main/java/com/sun/javafx/css/BitSet.java

```diff
@@ -539,31 +543,31 @@ public boolean equals(Object obj) {
     private boolean equalsBitSet(BitSet<?> other) {
         int a = this.bits != null ? this.bits.length : 0;
         int b = other.bits != null ? other.bits.length : 0;
+        int max = Math.max(a, b);
 
-        if (a != b) return false;
+        for (int i = 0; i < max; i++) {
+            long m0 = i >= a ? 0 : this.bits[i];
```

Comment:
Are we guaranteed that it is an invariant of this Class that there will never be an entry in `bits` that is `0`? From the code, it looks like that is the case (at least when inspecting add and remove), but it's not enforced I believe (e.g. subclasses may violate this rule). 
This remark is irrelevant if we consider 2 BitSets to be equal if their length is not equal, but the entries that are only in 1 of the sets are always 0.

Reply:
Note that this is rare anyway, these sets were rarely mutated (if ever), which is also the reason why this caching change was easy to do.  This is also why it never broke in practice.